pkg/declarativeconfig/access_scope.go

```diff
@@ -0,0 +1,57 @@
+package declarativeconfig
+
+import (
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"gopkg.in/yaml.v3"
+)
+
+type AccessScope struct {
+	Name        string `yaml:"name,omitempty"`
+	Description string `yaml:"description,omitempty"`
+	Rules       Rules  `yaml:"rules,omitempty"`
+}
+
+type Namespace struct {
+	Cluster   string `yaml:"cluster,omitempty"`
+	Namespace string `yaml:"namespace,omitempty"`
+}
+
+type Operator storage.SetBasedLabelSelector_Operator
+
+// MarshalYAML transforms Access to YAML format.
+func (a Operator) MarshalYAML() ([]byte, error) {
+	protoAccess := storage.SetBasedLabelSelector_Operator(a)
+	return []byte(protoAccess.String()), nil
+}
+
+// UnmarshalYAML makes transformation from YAML to Access.
+func (a *Operator) UnmarshalYAML(value *yaml.Node) error {
+	var v string
+	if err := value.Decode(&v); err != nil {
+		return err
+	}
+	i, ok := storage.SetBasedLabelSelector_Operator_value[v]
+	if !ok {
+		return errors.Errorf("Operator value %s not found", v)
+	}
+	*a = Operator(i)
+	return nil
+}
+
+type Requirement struct {
+	Key      string   `yaml:"key,omitempty"`
+	Operator Operator `yaml:"operator,omitempty"`
+	Values   []string `yaml:"values,omitempty"`
+}
+
+type LabelSelector struct {
+	Requirements []Requirement `yaml:"requirements,omitempty"`
+}
+
+type Rules struct {
+	IncludedClusters        []string        `yaml:"includedClusters,omitempty"`
+	IncludedNamespaces      []Namespace     `yaml:"includedNamespaces,omitempty"`
```

Comment:
It's a bit unfortunate now that I have to do the following for multiple namespaces from the same cluster:
```yaml
includedClusters:
- cluster A
includedNamespaces:
- cluster: cluster A
  namespace: namespace 1
- cluster: cluster A
  namespace: namespace 2
- cluster: cluster A
  namespace: namespace 2
```

This seems a bit suboptimal. I feel like we could do two things here: either allow to specify multiple namespaces per cluster, avoiding us to specify `cluster` multiple times _or_ we go one step further and merge both `includedClusters/includedNamespaces` together to something like:
```yaml
included:
- cluster: cluster A
  namespace:
  - ns 1
  - ns 2
  - ns 3
```
Then we have just a single setting that holds the included cluster / namespaces (with the usual caveat, that namespaces can be empty for each cluster)

Reply:
The second option sounds reasonable. It's hard for me to imagine a case where we would want explicitly specify cluster and 0 namespaces within it, so we can assume `cluster + 0 namespaces = the whole cluster`.